charts/akeyless-api-gateway/templates/serviceAccount.yaml

```diff
@@ -8,6 +9,42 @@ metadata:
   {{- if .annotations }}
   annotations:
     {{- toYaml .annotations | nindent 4 }}
-  {{- end -}}
-{{- end -}}
-{{- end -}}
\ No newline at end of file
+  {{- end }}
+
+  {{/* RBAC rule to access the secret if storeEncryptionKeyToK8sSecrets is true and clusterCacheEncryptionKeyExist is not empty */}}
+  {{- if and (eq $.Values.cachingConf.clusterCache.storeEncryptionKeyToK8sSecrets true) (not (eq (include "akeyless-api-gw.clusterCacheEncryptionKeyExist" $) "")) }}
```

Comment:
this condition needs to be update to support all the cases:

- if secret exist
- if secret not exist

should basically be create if cache enabled
@rani-sharim